lib/puppet/indirector/request.rb

```diff
@@ -217,11 +217,11 @@ def do_request(srv_service=:puppet, default_server=nil, default_port=nil, &block
       if primary_server = Puppet.settings[:server_list][0]
         bound_port = primary_server[1]
       else
-        bound_port = nil
+        bound_port = Puppet.settings[:masterport]
       end
     end
-    self.server = default_server || bound_server || Puppet.settings[:server]
-    self.port   = default_port || bound_port || Puppet.settings[:masterport]
+    self.server = default_server || bound_server
+    self.port   = default_port || bound_port
```

Comment:
We'll enter the rescue whenever Failover has not been attempted. This means one of two things

* `server_list` is unset
* We hit a codepath that doesn't involve the failover logic